packages/stargate/src/signingstargateclient.ts

```diff
@@ -346,7 +351,11 @@ export class SigningStargateClient extends StargateClient {
     if (!accountFromSigner) {
       throw new Error("Failed to retrieve account from signer");
     }
-    const pubkey = encodePubkey(encodeSecp256k1Pubkey(accountFromSigner.pubkey));
+    const pubkey = encodePubkey(
+      chainId.startsWith("evmos_9001-")
```

Comment:
Is there a more general approach over the evmos chain id? Other chains may be using the EthSecp keys from an ethermint integration. It would be nice if they can also use the cosm library in this case.

Reply:
When this is the case then we can make the code more complex so that it is manageable. No need to make it more complex for the moment as long as it is not useful.

Reply:
we already have other chains using the evm module from ethermint other than evmos. A few were mentioned here already (cronos, kava etc), and I am working on one myself. Given that I would love to see ethermint types supported in cosmjs in a way that these other chains could leverage.

Reply:
Oh I didn't know that. So I'm going to rework this so it's scalable and get back to you when it's right. Thanks for the heads up.